client/db.go

```diff
@@ -246,7 +246,16 @@ type Store interface {
 	GetAllIndexes(context.Context) (map[CollectionName][]IndexDescription, error)
 
 	// ExecRequest executes the given GQL request against the [Store].
-	ExecRequest(ctx context.Context, request string) *RequestResult
+	ExecRequest(ctx context.Context, request GQLRequest) *RequestResult
```

Comment:
thought: I am very reluctant to support coupling this function so strongly to a single query language.

Reply:
suggestion: Reworking this to use the option pattern would remove this coupling:

```
func WithOperation(operationName string) RequestOption {
  return func(o *gqlOptions) {
     o.OperationName = operationName
  }
}

ExecRequest(ctx context.Context, request string, opts ...RequestOption) *RequestResult
```

Reply:
I've pictured the abstraction to look like this:

```go
ExecRequest(ctx context.Context, request *request.Request) *RequestResult
```

The language frontend would handle converting the query into the `request.Request` format and the result would be converted by the frontend into the expected result format.

This approach should result in the removal of any GQL specific things from the DB interface, but will also require reformatting all of the tests to use `request.Request` instead of gql queries.

Reply:
I like this quite a lot and it would save embedded clients from a lot of parsing work.

I don't think the tests would need changing right away, initially you could just parse the gql strings into `request.Request` in the test framework (instead of internally as it is now).

The framework will always need to handle such a conversion anyway (to auto-test all query langs including the embedded), and although long term we may prefer writing most tests using the embedded, I think tests should be free to be written in any lang (and the test framework handle the conversion) - this would allow proper language specific testing.

Meaning we can have this in this PR without subjecting you to another really tedious reformat of all our tests.